src/optimizer/operators.cpp

```diff
@@ -553,6 +527,104 @@ Operator PhysicalLimit::make(int64_t offset, int64_t limit) {
   return Operator(limit_op);
 }
 
+//===--------------------------------------------------------------------===//
+// NLJoin (Inner + Outer Joins)
+//===--------------------------------------------------------------------===//
+Operator PhysicalNLJoin::make(
+    JoinType _type, std::vector<AnnotatedExpression> conditions,
+    std::vector<std::unique_ptr<expression::AbstractExpression>> &left_keys,
+    std::vector<std::unique_ptr<expression::AbstractExpression>> &right_keys) {
+  PhysicalNLJoin *join = new PhysicalNLJoin();
+  join->join_predicates = std::move(conditions);
+  join->left_keys = std::move(left_keys);
+  join->right_keys = std::move(right_keys);
+  join->type = _type;
+
+  return Operator(join);
+}
+
+hash_t PhysicalNLJoin::Hash() const {
+  hash_t hash = BaseOperatorNode::Hash();
+  for (auto &expr : left_keys)
```

Comment:
can these be for `(cost auto &...`)? Generally good to prefer using const where possible. Based on an initial glance looks like it could be but i may have missed something